src/universal_transfer_operator/data_providers/database/google/bigquery.py

```diff
@@ -155,6 +238,208 @@ def truncate_table(self, table):
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        native_support_kwargs = self.transfer_params.s3_transfer_parameters or {}
+
+        project_id = self.get_project_id(target_table)
+        transfer = S3ToBigqueryDataTransfer(
+            target_table=target_table,
+            source_file=source_file,
+            project_id=project_id,
+            native_support_kwargs=native_support_kwargs,
+            **kwargs,
+        )
+        transfer.run()
+
+    def get_project_id(self, target_table) -> str:
+        """
+        Get project id from the hook.
+        :param target_table: table object that the hook is derived from.
+        """
+        try:
+            return str(self.hook.project_id)
+        except AttributeError as exe:
+            raise DatabaseCustomError(f"conn_id {target_table.conn_id} has no project id") from exe
+
+    def load_local_file_to_table(
+        self,
+        source_file: File,
+        target_table: Table,
+        if_exists: LoadExistStrategy = "replace",
+        **kwargs,
+    ):
+        """
+        Transfer data from local to bigquery
+
+        :param source_file: Source file that is used as source of data
+        :param target_table: Table that will be created on the bigquery
+        :param if_exists: Overwrite table if exists. Default 'replace'
+        """
+        # We need to maintain file_type to biqquery_format and not use NATIVE_PATHS_SUPPORTED_FILE_TYPES
+        # because the load_table_from_file expects 'JSON' value for ndjson file.
+        file_types_to_bigquery_format = {
+            FileType.CSV: "CSV",
+            FileType.NDJSON: "JSON",
+            FileType.PARQUET: "PARQUET",
+        }
+
+        client = self.hook.get_client()
+        config = {
+            "source_format": file_types_to_bigquery_format[source_file.type.name],
+            "create_disposition": "CREATE_IF_NEEDED",
+            "write_disposition": BIGQUERY_WRITE_DISPOSITION[if_exists],
+            "autodetect": True,
+        }
+        job_config = bigquery.LoadJobConfig(**config)
+
+        # Deepsource pointed out - OWASP Top 10 2021 Category A01 - Broken Access Control
+        # and Category A05 - Security Misconfiguration. Which are not applicable in this
+        # case since user is always using there credentials, so they can't impersonate
+        # other user roles.
+
+        # We are passing mode='rb' even for text files since Bigquery
+        # complain and ask to open file in 'rb' mode
+        with open(source_file.path, mode="rb") as file:  # skipcq: PTC-W6004
+            job = client.load_table_from_file(
+                file,
+                job_config=job_config,
+                destination=self.get_table_qualified_name(target_table),
+            )
+        job.result()
```

Comment:
Same question regarding checking status of the run

Reply:
` job.result()` implementation waits till the transfer job is successful or raises an error when failed.